### PR TITLE
Split huge authReducer/Add logout button

### PR DESCRIPTION
**Pull request by zerobias**

### CODE REVIEW

**Review by goodmind — APPROVED** [review]

File: src/app/actions.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  DONE: createAction(`[end] ${reduxMessage}`),
+  FAIL: createAction(`[fail] ${reduxMessage}`),
+});
+
+export const AUTH = {
+  SEND_CODE: actionEvent('send code'),
+  SIGN_IN: actionEvent('sign in'),
+  GET_PASSWORD: actionEvent('get password'),
+  LOG_OUT: actionEvent('user log out'),
+};
```
Comment (goodmind) [inline src/app/actions.ts]: Move to `app/redux/modules`

File: src/app/components/Header/index.tsx [hunk 2]
```diff
@@ -4,16 +4,32 @@ import { Link } from 'react-router';
...
 const s = require('./style.css');
 
+import { logOut } from '../../redux/modules/auth';
+import { connect } from 'react-redux';
+
 class Header extends React.Component<any, any> {
+  constructor(...args) {
+    super(...args);
+    this.render = this.render.bind(this);
+  }
```
Comment (goodmind) [inline src/app/components/Header/index.tsx]: Why bind render?

File: src/app/redux/modules/auth/index.ts [hunk 3]
```diff
@@ -1,137 +1,82 @@
-import { IAuth, IAuthAction } from 'models/auth';
+// import { IAuth, IAuthAction, IAuthActions } from 'models/auth';
+// import { authKeyWithSaltToStorableBuffer } from 'helpers/Telegram';
```
Comment (goodmind) [inline src/app/redux/modules/auth/index.ts]: Why this is here?

**Review by goodmind — CHANGES_REQUESTED** [review]
Component changes is ok

File: src/app/actions.ts (hunk 1 above)
Comment (goodmind) [inline src/app/actions.ts]: Or just `app/redux`

File: src/app/redux/modules/auth/index.ts (hunk 3 above)
Comment (zerobias) [inline src/app/redux/modules/auth/index.ts]: Ok, my bad)

**Pull request merged**